tests/BotTest.php

```diff
@@ -781,3 +804,25 @@ public function handleEvent()
         // left empty for stubbing
     }
 }
+
+/**
+ * Plugin class which implements PluginContainerInterface.
+ */
+class TestPluginContainer extends AbstractPlugin implements PluginContainerInterface
```

Comment:
It should be possible to create the container as a mock of `PluginContainerInterface` with `getPlugins()` stubbed to return the plugin array rather than declaring a separate class.